pkg/binary/envoy/debug/log.go

```diff
@@ -75,22 +80,3 @@ func createLogFile(path string) (*os.File, error) {
 	}
 	return f, nil
 }
-
-func capture(r binary.Runner, file io.Closer) {
-	r.RegisterWait(1)
-	wait(r, binary.StatusTerminated)
-	if err := file.Close(); err != nil {
-		log.Errorf("error closing access log file: %v", err)
-	}
-	r.RegisterDone()
-}
-
-// wait blocks until the child process reaches the state passed
-// Note: It does not guarantee that it is in the specified state just that it has reached it
-func wait(r binary.Runner, state int) {
-	for r.Status() < state {
```

Comment:
the key change of this PR is moving from indirect inspection of process state, to direct. Anything like archiving logs should happen after the process terminates. It is easier to understand what's going on by looking where the process was started.